nodejs/eks/cluster.ts

```diff
@@ -951,7 +951,18 @@ export function createCore(
                             const getAnnosOutputStr = getAnnosOutput.toString();
                             // See if getAnnosOutputStr contains the annotation we're looking for.
                             if (!getAnnosOutputStr.includes("eks.amazonaws.com/compute-type") ) {
-                                // No need to patch, the annotation is not present.
+                                // No need to patch the deployment object since the annotation is not present. However, we need to re-create the CoreDNS pods since
+                                // the existing pods were created before the FargateProfile was created, and therefore will not have been scheduled by fargate-scheduler.
+                                // See: https://github.com/pulumi/pulumi-eks/issues/1030.
+                                const cmd = `kubectl rollout restart deployment coredns -n kube-system`;
```

Comment:
Did not expect to find this change there, curious is there any alternative to shelling out? What if the host system does not have kubectl installed?

Reply:
We have existing logic that already shells out to `kubectl` for updating the existing CoreDNS deployment object: https://github.com/pulumi/pulumi-eks/blob/62b0c2c788ec60715e6e139d85c570988e4b2ee2/nodejs/eks/cluster.ts#L944-L971
The additional code introduced in this PR simply builds upon that existing functionality.

Fwiw we do check that `kubectl` exist on the local system, and that it is a compatible version as well: https://github.com/pulumi/pulumi-eks/blob/62b0c2c788ec60715e6e139d85c570988e4b2ee2/nodejs/eks/dependencies.ts#L31-L62
Associated tests for the kubectl checking logic: https://github.com/pulumi/pulumi-eks/blob/62b0c2c788ec60715e6e139d85c570988e4b2ee2/nodejs/eks/dependencies.test.ts

Kubernetes does provide generated clients for NodeJS (https://github.com/kubernetes-client/javascript), so it might be nice to use that in the future for these operations.